src/BackgroundLocationTracking/MainPage.xaml.cs

```diff
@@ -0,0 +1,179 @@
+#endif
+                await StartLocationDataSource();
+            }
+            catch (Exception ex)
+            {
+                await Shell.Current.DisplayAlert("Error", $"An error occurred while starting tracking: {ex.Message}.\n Please try again.", "OK");
+                _isTracking = false;
+            }
+            UpdateButtonStates();
+        }
+
+        private async void StopTracking(object sender, EventArgs e)
+        {
+            if (!_isTracking) return;
+
+            _isTracking = false;
+            UpdateButtonStates();
+
+            try
+            {
+#if ANDROID
+                var intent = new Android.Content.Intent(Android.App.Application.Context, typeof(LocationService));
+
+                // Stop the foreground service when tracking is stopped.
+                Android.App.Application.Context.StopService(intent);
+#endif
+                await StopLocationDataSource();
+            }
+            catch (Exception ex)
+            {
+                await Shell.Current.DisplayAlert("Error", $"An error occurred while stopping tracking: {ex.Message}.\n Please try again.", "OK");
+                _isTracking = true;
+            }
+            UpdateButtonStates();
+        }
+
+        private void UpdateButtonStates()
+        {
+            StartButton.IsEnabled = !_isTracking;
+            StopButton.IsEnabled = _isTracking;
+        }
+
+        private async Task StartLocationDataSource()
+        {
+            // Clear previous trail before starting the location data source.
+            ClearTrail();
+            if (_locationDataSource is not null)
+            {
+                await _locationDataSource.StartAsync();
+
+                MyMapView.LocationDisplay.DataSource = _locationDataSource;
+                MyMapView.LocationDisplay.IsEnabled = true;
+                MyMapView.LocationDisplay.InitialZoomScale = 1000;
+                MyMapView.LocationDisplay.AutoPanMode = LocationDisplayAutoPanMode.Recenter;
+            }
+        }
+
+        private async Task StopLocationDataSource()
+        {
+            if (_locationDataSource is not null)
+            {
+                await _locationDataSource.StopAsync();
+                MyMapView.LocationDisplay.IsEnabled = false;
+            }
+        }
+
+        private void LocationDataSource_LocationChanged(object? sender, Location e)
+        {
+            if (e is not null)
+            {
+                MainThread.BeginInvokeOnMainThread(() =>
+                {
+                    var projectedPoint = (MapPoint)GeometryEngine.Project(e.Position, SpatialReferences.Wgs84);
+                    _polylineBuilder?.AddPoint(projectedPoint);
+
+                    var geometry = _polylineBuilder?.ToGeometry();
+                    _trackLineOverlay?.Graphics.Clear();
+                    _trackLineOverlay?.Graphics.Add(new Graphic(geometry));
+                    MyMapView.SetViewpointGeometryAsync(geometry!, 50);
```

Comment:
You have autopanmode set, so I'd think this shouldn't be necessary and would just disable the autopan?